systems-state-manager/nisystemsstate.yml

```diff
@@ -0,0 +1,999 @@
+      summary: API version information
+      description: Returns available operations for a single version of the API.
+      tags: [versioning]
+      security: []
+      parameters:
+        - in: path
+          name: version
+          description: The version of the API to retrieve operations.
+          x-example: v1
+          type: string
+          required: true
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/V1Operations'
+        404:
+          description: Not Found
+        default:
+            $ref: '#/responses/Error'
+  /v1/states:
+    get:
+      operationId: GetStates
+      x-ni-privilege: Read
+      x-ni-operation: getStates
+      tags: [states]
+      summary: Get all system states
+      description: Return metadata information of all system states.
+      parameters:
+        - in: query
+          name: skip
+          description: How many states to skip. Used for pagination.
+          type: integer
+          x-ni-data-type: U32
+          default: 0
+        - in: query
+          name: take
+          description: How many states to retrieve. Used for pagination.
+          type: integer
+          x-ni-data-type: U32
+          default: 1000
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/StateDescriptionListResponse'
+        401:
+          $ref: '#/responses/Unauthorized'            
+        default:
+          $ref: '#/responses/Error'
+    post:
+      operationId: CreateState
+      x-ni-privilege: nisystemsstate.Write
+      x-ni-operation: createOrUpdateStates
+      tags: [states]
+      summary: Create a new system state
+      description: Create a new System State by providing metadata and its content. rawState property will be used if it has a non-null value, feeds and packages will be used otherwise.
+      parameters:
+        - in: body
+          name: state
+          description: The state to create.
+          required: true
+          schema:
+            $ref: '#/definitions/StateRequest'
+      responses:
+        201:
+          description: Created
+          schema:
+            $ref: '#/definitions/StateResponse'
+        400:
+          $ref: '#/responses/BadRequest'            
+        401:
+          $ref: '#/responses/Unauthorized'
+        409:
+          $ref: '#/responses/Conflict'
+        default:
+          $ref: '#/responses/Error'
+  /v1/states/{id}:
+    get:
+      operationId: GetState
+      x-ni-privilege: Read
+      x-ni-operation: getStates
+      tags: [states]
+      summary:  Get system state
+      description: Returns the metadata, the feeds and packages from the content of the state. If the state contains only feeds and packages entries the containsExtraOperations from response will be false, otherwise it will be true. The export route could be used in order to get the whole content of the state.
```

Comment:
"Returns the metadata, feeds, and packages from the state content. If the state contains only feed and package entries, the containsExtraOperations property of the response will be false. Otherwise, it will be true. You may use the export route to get all state content."?